### PR TITLE
Creation of MaterialEndofLifeTreatment method

### PR DESCRIPTION
**Pull request by michaelhoehn**
### Issues addressed by this PR
Addresses issue #26 

### Test files
ResourceManagement

### CODE REVIEW

File: ClimateEmergency_Engine/Compute/MaterialEndOfLifeTreatment.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    public static partial class Compute
+    {
+        /***************************************************/
+        /****   Public Methods                          ****/
+        /***************************************************/
+
+        [Description("Returns End of Life processing information contained within an EPD dataset.")]
+        [Input("EPD Data", "Environmental Product Declaration of a specific material from an EPD Dataset.")]
+        [Output("EOL Treatment", "End of Life treatment per material. This includes all data collected for LCA stages C1-C4 within a provided EPD dataset.")]
+
```
Comment (FraserGreenroyd) [inline ClimateEmergency_Engine/Compute/MaterialEndOfLifeTreatment.cs]: Remove blank line

File: ClimateEmergency_Engine/Compute/MaterialEndOfLifeTreatment.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        public static string MaterialEndofLifeTreatment(CustomObject EPDdata)
```
Comment (FraserGreenroyd) [inline ClimateEmergency_Engine/Compute/MaterialEndOfLifeTreatment.cs]: Input variables should start with lowercase letter. Would recommend `epdData`.

File: ClimateEmergency_Engine/Compute/MaterialEndOfLifeTreatment.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+namespace BH.Engine.ClimateEmergency
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /****   Public Methods                          ****/
+        /***************************************************/
+
+        [Description("Returns End of Life processing information contained within an EPD dataset.")]
+        [Input("EPD Data", "Environmental Product Declaration of a specific material from an EPD Dataset.")]
```
Comment (FraserGreenroyd) [inline ClimateEmergency_Engine/Compute/MaterialEndOfLifeTreatment.cs]: Input attribute name should match input variable name, see comment below about capitalisation but would recommend `[Input("epdData", "...")]`

File: ClimateEmergency_Engine/Compute/MaterialEndOfLifeTreatment.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            string treatmentEOL;
+
+            if (EPDdata.CustomData.ContainsKey("TreatmentEOL"))
+            {
+                return (string)EPDdata.CustomData["TreatmentEOL"];
+            }
+            else
+            {
+                BH.Engine.Reflection.Compute.RecordError("The EPDDataset must have a valid value for Treatment EOL under a 'TreatmentEOL' key.");
+                return (string)"No data to record. Please provide missing data within 'TreatmentEOL' key.";
```
Comment (FraserGreenroyd) [inline ClimateEmergency_Engine/Compute/MaterialEndOfLifeTreatment.cs]: This casting is unnecessary, suggest removing.

**Comment by kristenbrozowski** [comment]
image

Builds fine. Clever method!

File: ClimateEmergency_Engine/Compute/MaterialEndOfLifeTreatment.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        /***************************************************/
+        /****   Public Methods                          ****/
+        /***************************************************/
+
+        [Description("Returns End of Life processing information contained within an EPD dataset.")]
+        [Input("epdData", "Environmental Product Declaration of a specific material from an EPD Dataset.")]
+        [Output("EOL Treatment", "End of Life treatment per material. This includes all data collected for LCA stages C1-C4 within a provided EPD dataset.")]
+        public static string MaterialEndofLifeTreatment(CustomObject epddata)
+        {
+            string treatmentEOL;
```
Comment (FraserGreenroyd) [inline ClimateEmergency_Engine/Compute/MaterialEndOfLifeTreatment.cs]: If this variable isn't being used we can remove it.

File: ClimateEmergency_Engine/Compute/MaterialEndOfLifeTreatment.cs [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /****   Public Methods                          ****/
+        /***************************************************/
+
+        [Description("Returns End of Life processing information contained within an EPD dataset.")]
+        [Input("epdData", "Environmental Product Declaration of a specific material from an EPD Dataset.")]
+        [Output("EOL Treatment", "End of Life treatment per material. This includes all data collected for LCA stages C1-C4 within a provided EPD dataset.")]
```
Comment (FraserGreenroyd) [inline ClimateEmergency_Engine/Compute/MaterialEndOfLifeTreatment.cs]: Output name should be lowercase first letter and camel cased as with input names - I'd recommend `eolTreatment`.

**Pull request merged**